### PR TITLE
[chore] Enable goleak for tests failing on opencensus-go

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

This enables `goleak` checks for all packages whose only failure is due to [a known issue](https://github.com/census-instrumentation/opencensus-go/issues/1191) within the `opencensus-go` package. I believe it's safe to ignore for two main reasons:
1. All references that I'm aware of are indirect, the contrib repo isn't directly using the stat's worker functionality. The goroutine is started in the dependency's `init()` so it's not something that we're causing.
2. We're actively moving away from the opencensus dependency, and hope to remove it entirely soon. Relevant issue: https://github.com/open-telemetry/opentelemetry-collector/issues/816. Once this work is complete the ignore will no longer be necessary and we can safely remove it.

The only changes in this PR are test related. The same file is being copied into different packages, and then we run `make gotidy`. No functionality has been changed.

**Link to tracking Issue:** <Issue number if applicable>
#30438

**Testing:** <Describe what testing was performed and which tests were added.>
All added tests are passing.

### CODE REVIEW

**Comment by crobert-1** [comment]
I'm going to wait on this until the #30451 is merged. This PR is essentially stacked on top of that one. Changes in `main` are changing the scope of #30451, which in turn changes the scope of this. It's a lot of extra work to maintain them in parallel, so I'll just wait for that one to be complete.

**Comment by crobert-1** [comment]
Alright, this should be ready to review, apologies again for the size. Refer to [this comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30451#issuecomment-1896766297) for a simple way to ensure the changes are just one copy being filed into numerous placeas.

**Review by fatsheep9146 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
